src/Symfony/Component/Cache/Adapter/TagAwareAdapter.php

```diff
@@ -314,7 +288,31 @@ public function saveDeferred(CacheItemInterface $item)
      */
     public function commit()
     {
-        return $this->invalidateTags([]);
+        if (!$this->deferred) {
+            return true;
+        }
+
+        $ok = true;
+        $tagsByKey = [];
+        $items = $this->deferred;
+        foreach ($items as $key => $item) {
+            if (!$this->pool->saveDeferred($item)) {
+                unset($this->deferred[$key]);
+                $ok = false;
+            }
+        }
+
+        $tagsByKey = (self::$getTagsByKey)($items);
```

Comment:
How likey is it to get `false` from `saveDeffered()`?
Such possibility itself implies that items may be committed partially while their tag versions will always be updated.
Wouldn't it be safer to extract tags only from succesfully deferred items?
`$tagsByKey = (self::$getTagsByKey)($this->deferred);`

Reply:
right, updated (probability is very very low, it's even zero in Symfony since saveDeferred is a local operation)